python/ray/_private/function_manager.py

```diff
@@ -124,15 +118,37 @@ def compute_collision_identifier(self, function_or_class):
         # Return a hash of the identifier in case it is too large.
         return hashlib.sha1(collision_identifier.encode("utf-8")).digest()
 
+    def check_local(self,module_name,function_class_name):
+        """Check if a function or class in the module is local."""
+        module = importlib.import_module(module_name)
+        parts = [part for part in function_class_name.split(".") if part]
+        object = module
+        try:
+            for part in parts:
+                object = getattr(object, part)
+            return object
+        except:
+            return
+        
     def export(self, remote_function):
         """Pickle a remote function and export it to redis.
 
         Args:
             remote_function: the RemoteFunction object.
         """
-        if self._worker.load_code_from_local:
-            return
 
+        function_descriptor = remote_function._function_descriptor
+        if function_descriptor.is_actor_method():
+            return
+        function_id = function_descriptor.function_id
+        if (function_id in self._function_execution_info):
+            return
+        module_name, function_name = (
+            function_descriptor.module_name,
+            function_descriptor.function_name,
+        )
+        if self.check_local(module_name,function_name):
```

Comment:
Please add some comments in each if statement. E.g., "If the function is an actor method, no need to export it to GCS".